packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2120,3 +2130,146 @@ class _Editable extends LeafRenderObjectWidget {
       ..paintCursorAboveText = paintCursorAboveText;
   }
 }
+
+// This formatter inserts [Unicode.RLM] and [Unicode.LRM] into the
+// string in order to preserve expected caret behavior when trailing
+// whitespace is inserted.
+//
+// When typing in a direction that opposes the base direction
```

Comment:
The first sentence isn't clear to me after reading. "un-enclosed whitespace gets the directionality of the paragraph"  sounds like that's a generally desired behavior. "When typing in a direction that opposes the base direction of the paragraph" maybe describe how you can cause this.